// react-text-effects/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TextTypeWriter = exports.TextShadow = exports.TextPopUp = exports.TextBlur = exports.TextFocus = undefined;

var _templateObject = _taggedTemplateLiteral(['\n  -webkit-animation: ', ' ', ' cubic-bezier(0.550, 0.085, 0.680, 0.530) both;\n  animation: ', ' ', ' cubic-bezier(0.550, 0.085, 0.680, 0.530) both;\n'], ['\n  -webkit-animation: ', ' ', ' cubic-bezier(0.550, 0.085, 0.680, 0.530) both;\n  animation: ', ' ', ' cubic-bezier(0.550, 0.085, 0.680, 0.530) both;\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  -webkit-animation: ', ' ', ' both;\n  animation: ', ' ', ' both;\n'], ['\n  -webkit-animation: ', ' ', ' both;\n  animation: ', ' ', ' both;\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n    width: 12em;\n    border-right: 2px solid black;\n    font-size: 180%;\n    text-align: center;\n    white-space: nowrap;\n    overflow: hidden;\n    animation: ', ' ', ' steps(34) 1s 1 normal both,\n         ', ' 500ms steps(34) infinite normal;\n'], ['\n    width: 12em;\n    border-right: 2px solid black;\n    font-size: 180%;\n    text-align: center;\n    white-space: nowrap;\n    overflow: hidden;\n    animation: ', ' ', ' steps(34) 1s 1 normal both,\n         ', ' 500ms steps(34) infinite normal;\n']);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _keyframes = require('./styles/keyframes');

var _string = require('./helpers/string');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var focusKlasses = { KfTextFocusIn: _keyframes.KfTextFocusIn, KfTextFocusExpand: _keyframes.KfTextFocusExpand, KfTextFocusContract: _keyframes.KfTextFocusContract };
var blurKlasses = { KfTextBlurOut: _keyframes.KfTextBlurOut, KfTextBlurContract: _keyframes.KfTextBlurContract, KfTextBlurExpand: _keyframes.KfTextBlurExpand };
var popKlasses = { KfTextPopUpTop: _keyframes.KfTextPopUpTop, KfTextPopUpRight: _keyframes.KfTextPopUpRight, KfTextPopUpBottom: _keyframes.KfTextPopUpBottom, KfTextPopUpLeft: _keyframes.KfTextPopUpLeft };
var shadowKlasses = { KfShadowDropCenter: _keyframes.KfShadowDropCenter, KfShadowDropLeft: _keyframes.KfShadowDropLeft, KfShadowDropRight: _keyframes.KfShadowDropRight, KfShadowDropTop: _keyframes.KfShadowDropTop, KfShadowDropBottom: _keyframes.KfShadowDropBottom };

var textFocusType = function textFocusType(type) {
  return focusKlasses['KfTextFocus' + (0, _string.capitalize)(type)];
};

var blurType = function blurType(type) {
  return blurKlasses['KfTextBlur' + (0, _string.capitalize)(type)];
};

var popType = function popType(type) {
  return popKlasses['KfTextPopUp' + (0, _string.capitalize)(type)];
};

var shadowType = function shadowType(type) {
  return shadowKlasses['KfShadowDrop' + (0, _string.capitalize)(type)];
};

var TextFocus = exports.TextFocus = _styledComponents2.default.div(_templateObject, function (props) {
  return textFocusType(props.type);
}, function (props) {
  return props.duration || '1s';
}, function (props) {
  return textFocusType(props.type);
}, function (props) {
  return props.duration || '1s';
});

var TextBlur = exports.TextBlur = _styledComponents2.default.div(_templateObject, function (props) {
  return blurType(props.type);
}, function (props) {
  return props.duration || '1s';
}, function (props) {
  return blurType(props.type);
}, function (props) {
  return props.duration || '1s';
});

var TextPopUp = exports.TextPopUp = _styledComponents2.default.div(_templateObject, function (props) {
  return popType(props.type);
}, function (props) {
  return props.duration || '1s';
}, function (props) {
  return popType(props.type);
}, function (props) {
  return props.duration || '1s';
});

var TextShadow = exports.TextShadow = _styledComponents2.default.div(_templateObject2, function (props) {
  return shadowType(props.type);
}, function (props) {
  return props.duration || '1s';
}, function (props) {
  return shadowType(props.type);
}, function (props) {
  return props.duration || '1s';
});

var TextTypeWriter = exports.TextTypeWriter = _styledComponents2.default.div(_templateObject3, _keyframes.KfTypewriter, function (props) {
  return props.duration || '1s';
}, _keyframes.KfBlinkTextCursor);
